src/gfx/process.cpp

```diff
@@ -736,6 +736,20 @@ class TileData {
 		return row;
 	}
 
+	TileData(std::array<uint8_t, 16> &&raw) : _data(raw), _hash(0) {
+		for (uint8_t y = 0; y < 8; ++y) {
+			uint16_t bitplanes = _data[y * 2] | _data[y * 2 + 1] << 8;
+
+			_hash ^= bitplanes;
+			if (options.allowMirroring) {
+				// Count the line itself as mirrorred; vertical mirroring is
+				// already taken care of because the symmetric line will be XOR'd
+				// the same way. (...which is a problem, but probably benign.)
```

Comment:
Having this exact long comment copy+pasted from elsewhere feels suspicious to me. Either the logic should be factored out, or the "probably benign" problem should be dealt with if it's now spreading.

(Why do we even hash the tiles, anyway? 8x8px \* 2bpp / 8 bits/byte = 16 bytes per tile; just store/compare that data directly.

Reply:
We hash them because the C++ hashing interface returns `size_t` and nothing else, so we need to reduce to that somehow.

I don't think the stated problem is relevant with most input data, but I can factor out the function, if you want.

Reply:
Yeah, I'm not asking to change/remove the hashing in this PR (it'd be a separate PR even if we did so at all), but factoring this out would be great, please.

Reply:
And thank you for explaining why we need to hash in the first place (C++ STL API).